.github/workflows/release.yml

```diff
@@ -0,0 +1,161 @@
+name: Release
+
+on:
+  workflow_dispatch:
+
+permissions:
+  contents: write
+
+jobs:
+  get-version:
+    name: Get version from Cargo.toml
+    runs-on: ubuntu-latest
+    outputs:
+      version: ${{ steps.version.outputs.version }}
+
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Get version from Cargo.toml
+        id: lookupVersion
+        uses: mikefarah/yq@dd648994340a5d03225d97abf19c9bf1086c3f07
+        with:
+          cmd: yq -oy '"v" + .package.version' 'Cargo.toml'
+
+      - name: Print version
+        id: version
+        run: |
+          VERSION=${{ steps.lookupVersion.outputs.result }}
+          echo "version=$VERSION" >> "$GITHUB_OUTPUT"
+
+  build-binaries:
+    name: Build ${{ matrix.target }}
+    continue-on-error: true
+    needs: get-version
+    runs-on: ${{ matrix.os }}
+
+    env:
+      # Cross-compiled targets will override this to `cross`.
+      CARGO: cargo
+
+    strategy:
+      matrix:
+        include:
+          - target: x86_64-unknown-linux-gnu
+            os: ubuntu-latest
+            cross: true
+
+          - target: x86_64-unknown-linux-musl
+            os: ubuntu-latest
+            cross: true
+
+          - target: aarch64-unknown-linux-gnu
+            os: ubuntu-latest
+            cross: true
+
+          - target: aarch64-unknown-linux-musl
+            os: ubuntu-latest
+            cross: true
+
+          - target: x86_64-apple-darwin
+            os: macos-latest
+
+          - target: aarch64-apple-darwin
+            os: macos-latest
+
+          - target: x86_64-pc-windows-msvc
+            os: windows-latest
+
+    steps:
+      - uses: actions/checkout@v4
+
+      - uses: dtolnay/rust-toolchain@439cf607258077187679211f12aa6f19af4a0af7
+        with:
+          toolchain: stable
+          target: ${{ matrix.target }}
+
+      - uses: Swatinem/rust-cache@23bce251a8cd2ffc3c1075eaa2367cf899916d84
+
+      - name: Install cross
+        if: matrix.cross
+        uses: taiki-e/install-action@cross
+
+      - name: Enable cross-compilation
+        if: matrix.cross
+        shell: bash
+        run: |
+          echo "CARGO=cross" >> $GITHUB_ENV
+
+      - uses: actions/setup-go@v5
+      - uses: arduino/setup-protoc@v3
+
+      - name: Configure cross and build
+        continue-on-error: true
```

Comment:
nit: what's the rationale behind this here?

Reply:
It was nice when I was debugging. I'll remove it because we do not want release to pass without some artifacts